lightning/src/ln/channelmanager.rs

```diff
@@ -2903,192 +2920,28 @@ where
 		}
 	}
 
-	fn construct_fwd_pending_htlc_info(
-		&self, msg: &msgs::UpdateAddHTLC, hop_data: msgs::InboundOnionPayload, hop_hmac: [u8; 32],
-		new_packet_bytes: [u8; onion_utils::ONION_DATA_LEN], shared_secret: [u8; 32],
-		next_packet_pubkey_opt: Option<Result<PublicKey, secp256k1::Error>>
-	) -> Result<PendingHTLCInfo, InboundOnionErr> {
-		debug_assert!(next_packet_pubkey_opt.is_some());
-		let outgoing_packet = msgs::OnionPacket {
-			version: 0,
-			public_key: next_packet_pubkey_opt.unwrap_or(Err(secp256k1::Error::InvalidPublicKey)),
-			hop_data: new_packet_bytes,
-			hmac: hop_hmac,
-		};
-
-		let (short_channel_id, amt_to_forward, outgoing_cltv_value) = match hop_data {
-			msgs::InboundOnionPayload::Forward { short_channel_id, amt_to_forward, outgoing_cltv_value } =>
-				(short_channel_id, amt_to_forward, outgoing_cltv_value),
-			msgs::InboundOnionPayload::Receive { .. } | msgs::InboundOnionPayload::BlindedReceive { .. } =>
-				return Err(InboundOnionErr {
-					msg: "Final Node OnionHopData provided for us as an intermediary node",
-					err_code: 0x4000 | 22,
-					err_data: Vec::new(),
-				}),
-		};
-
-		Ok(PendingHTLCInfo {
-			routing: PendingHTLCRouting::Forward {
-				onion_packet: outgoing_packet,
-				short_channel_id,
-			},
-			payment_hash: msg.payment_hash,
-			incoming_shared_secret: shared_secret,
-			incoming_amt_msat: Some(msg.amount_msat),
-			outgoing_amt_msat: amt_to_forward,
-			outgoing_cltv_value,
-			skimmed_fee_msat: None,
-		})
-	}
-
 	fn construct_recv_pending_htlc_info(
```

Comment:
Can't we just get rid of this method now, since the only thing it does is call a different method?

Reply:
ok ya, i just got rid of it and call `create_recv_pending_htlc_info` from the tests too